### PR TITLE
Fix the checksum verification to be more sensitive

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 1]
```diff
@@ -39,12 +39,14 @@ class UpdatePackagesCommand extends FlutterCommand {
         help: 'Attempt to update all the dependencies to their latest versions.\n'
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/update_packages.dart]: all actual changes are in this file

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/update_packages.dart]: Originally were a workaround for dart 1 and dart 2 support, can probably be removed now

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by Hixie** [comment]
@jonahwilliams Can you PTAL, I changed a bunch of the logic some more. (There's a separate commit for the changes to the code vs the changes to the `pubspec.yaml`s.)

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 2]
```diff
@@ -163,7 +169,12 @@ class UpdatePackagesCommand extends FlutterCommand {
       final Set<String> specialDependencies = new Set<String>();
       for (Directory directory in packages) { // these are all the directories with pubspec.yamls we care about
         printTrace('Reading pubspec.yaml from: ${directory.path}');
-        final PubspecYaml pubspec = new PubspecYaml(directory); // this parses the pubspec.yaml
+        PubspecYaml pubspec;
+        try {
+          pubspec = new PubspecYaml(directory); // this parses the pubspec.yaml
+        } on String catch (message) {
+          throwToolExit(message);
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/commands/update_packages.dart]: Why not call throwToolExit in `PubspecYaml._parse` ?

**Review by jonahwilliams — APPROVED** [review]
LGTM - bot failures look like flakes

**Comment by Hixie** [comment]
Turns out, not flakes. Need to pin flutter_gallery_assets.

**Pull request merged**